lib/mix/test/test_helper.exs

```diff
@@ -412,6 +412,21 @@ defmodule Mix.Tasks.Hello do
 end
 |> write_beam.()
 
+defmodule Mix.Tasks.WithRequirement do
+  use Mix.Task
+  @shortdoc "This is short documentation, see"
+  @requirements ["help compile"]
+
+  @moduledoc """
+  A test task.
+  """
+
+  def run(_args) do
+    "Task with requirements"
+  end
+end
+|> write_beam.()
```

Comment:
I think you can define it within the `mix/task_test.exs`, without the `write_beam.()` code, and it should all work. Have you tried it?

Reply:
It works but the code must be outside `Mix.TaskTest` module definition, otherwise the namespace gets messed up. It creates de module under `Mix.TaskTest.Mix.Tasks.WithRequirement`.

It doesn't seem natural to have this module definition under the same file of `Mix.TaskTest`, but outside its definition. Any suggestions?

Reply:
You can move it to the top of the file or name it `defmodule Elixir.Mix.Tasks.WithRequirement do`. :)